iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -73,6 +73,21 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
     auto m_segmentInfo = m_segmentManager->getSegmentInformationForUser(currentUser);
     m_memoryManagerOfCurrentProcess = m_segmentInfo.m_memoryManager;
 }
+
+void ProcessManager::handleProcessShutdownPreparationRequest(const RuntimeName_t& name) noexcept
+{
+    searchForProcessAndThen(
+        name,
+        [&](Process& process) {
+            m_portManager.unblockProcessShutdown(name);
+            // Reply with PREPARE_APP_TERMINATION_ACK and let process shutdown
+            runtime::IpcMessage sendBuffer;
+            sendBuffer << runtime::IpcMessageTypeToString(runtime::IpcMessageType::PREPARE_APP_TERMINATION_ACK);
+            process.sendViaIpcChannel(sendBuffer);
+        },
+        [&]() { LogWarn() << "Unknown application " << name << " requested shutdown preparation."; });
```

Comment:
Is this really level LogWarn?

Reply:
Well, when this happens, someone is tampering with RouDi by opening the roudi socket and sending arbitrary messages. The same log level is used in the other cases so I copied it.